### PR TITLE
proper resources handling

### PR DESCRIPTION
**Pull request by jbleduigou**
Signed-off-by: jbleduigou <jb.leduigou@gmail.com>

### CODE REVIEW

File: common-app-api/src/main/java/org/openecomp/sdc/common/http/client/api/HttpRequestHandler.java [hunk 1]
```diff
@@ -79,13 +79,27 @@ public static void setTestInstance(HttpRequestHandler handlerInstance) {
     }
 
     public HttpResponse<byte[]> getAsByteArray(String url, Properties headers, HttpClientConfig config) throws HttpExecuteException {
-        HttpClient client = getOrCreateClient(url, config);
-        return client.<byte[]>get(url, headers, byteResponseBuilder);
+        HttpClient client = null;
+        try {
```
Comment (devfest-bugbust) [inline common-app-api/src/main/java/org/openecomp/sdc/common/http/client/api/HttpRequestHandler.java:83]: What about using try with resources ? `try(HttpClient client = getOrCreateClient(url, config)) {` ?

**Review by devfest-bugbust — CHANGES_REQUESTED** [review]

File: common-app-api/src/main/java/org/openecomp/sdc/common/http/client/api/HttpRequestHandler.java (hunk 1 above)
Comment (jbleduigou) [reply common-app-api/src/main/java/org/openecomp/sdc/common/http/client/api/HttpRequestHandler.java:83]: Yes I thought about that too.
Unfortunately the class `HttClient` is a custom one which does not implement the `AutoCloseable` interface.

Comment (jbleduigou) [reply common-app-api/src/main/java/org/openecomp/sdc/common/http/client/api/HttpRequestHandler.java:83]: Yes I thought about that too.
Unfortunately the class `HttClient` is a custom one which does not implement the `AutoCloseable` interface.

Comment (devfest-bugbust) [reply common-app-api/src/main/java/org/openecomp/sdc/common/http/client/api/HttpRequestHandler.java:83]: Fair enough LGTM

**Review by devfest-bugbust — APPROVED** [review]

**Pull request merged**